trace/HTTP.md

```diff
@@ -90,20 +103,22 @@ Implementations SHOULD create message event when response size is determined.
 ## Attributes
 
 Implementations SHOULD set the following attributes on the client and server spans. For a server,
- request represents the incoming request. For a client, request represents the outgoing request.
-
-All attributes are optional.
-
-| Attribute name            | Description                 | Example value                   |
-|---------------------------|-----------------------------|---------------------------------|
-| "http.host"               | Request URL host            | "example.com:779"               |
-| "http.method"             | Request URL method          | "GET"                           |
-| "http.path"               | Request URL path            | "/users/25f4c31d"               |
-| "http.route"              | Matched request URL route   | "/users/:userID"                |
-| "http.user_agent"         | Request user-agent          | "HTTPClient/1.2"                |
-| "http.status_code"        | Response status code        | 200                             |
-
-Exporters should always export the collected attributes. Exporters should map the collected 
+request represents the incoming request. For a client, request represents the outgoing request.
+
+All attributes are optional, but collector should make the best effort to
+collect those.
+
+| Attribute name            | Description                 | Type   |Example value              |
+|---------------------------|-----------------------------|--------|---------------------------|
+| "http.host"               | Request URL host            | string | `example.com:779`         |
+| "http.method"             | Request URL method          | string | `GET`                     |
+| "http.path"               | Request URL path. If empty - set to `/` | `/users/25f4c31d`  |
+| "http.route"              | Matched request URL route   | string | `/users/:userID`          |
+| "http.user_agent"         | Request user-agent. Do not inject attribute if user-agent is empty. | string | `HTTPClient/1.2` |
+| "http.status_code"        | Response status code        | int64  | `200`                     |
+| "http.url"                | Absolute request URL        | string | `https://example.com:779/path/12314/?q=ddds#123` |
```

Comment:
Should we specify how this URL is to be constructed from the request (host header, protocol, etc.)?